src/Campaigns/resources/admin/components/MergeCampaign/Form/index.tsx

```diff
@@ -0,0 +1,308 @@
+        defaultValues: {
+            destinationCampaignId: '',
+        },
+    });
+
+    const {
+        register,
+        handleSubmit,
+        formState: {isDirty, isSubmitting},
+        watch,
+    } = methods;
+
+    const destinationCampaignId = watch('destinationCampaignId');
+
+    const getFormModalTitle = () => {
+        if (4 === step) {
+            return '' + title;
+        }
+
+        return title;
+    };
+
+    const requiredAsterisk = <span className={`givewp-field-required ${styles.fieldRequired}`}>*</span>;
+
+    const onSubmit: SubmitHandler<MergeCampaignFormInputs> = async (inputs, event) => {
+        event.preventDefault();
+
+        if (step !== 2) {
+            return;
+        }
+
+        const campaignsToMergeIds = campaigns.selected.filter((id) => id != inputs.destinationCampaignId);
+
+        try {
+            const response = await apiFetch({
+                path: addQueryArgs('/give-api/v2/campaigns/' + destinationCampaignId + '/merge', {
+                    campaignsToMergeIds: campaignsToMergeIds,
+                }),
+                method: 'PATCH',
+            });
+
+            console.log('Merge campaigns response: ', response);
+
+            // Go to success page
+            setStep(3);
+
+            //Reset bulk actions selector
+            const selects = document.querySelectorAll('#give-admin-campaigns-root select');
+            selects.forEach((select) => {
+                const selectElement = select as HTMLSelectElement;
+                selectElement.selectedIndex = 0;
+            });
+
+            // Uncheck all checkboxes
+            const checkboxes = document.querySelectorAll(".giveListTable input[type='checkbox']");
+            checkboxes.forEach((checkbox) => {
+                const input = checkbox as HTMLInputElement;
+                input.checked = false;
+            });
+            // @ts-ignore
+            document.querySelector('.giveListTable #giveListTableSelectAll').checked = false;
+
+            //Remove campaignsToMergeIds from the list table.
+            const adminFormsListViewItems = document.querySelectorAll('tr');
+            if (adminFormsListViewItems.length > 0) {
+                adminFormsListViewItems.forEach((itemElement) => {
+                    const select = itemElement.querySelector('.giveListTableSelect');
+
+                    if (!select) {
+                        return;
+                    }
+
+                    const campaignId = select.getAttribute('data-id');
+                    if (campaignsToMergeIds.includes(campaignId)) {
+                        itemElement.remove();
+                    }
+                });
+            }
+            //handleClose(response);
+        } catch (error) {
+            // Go to error page
+            setStep(4);
+            console.error('Error merging campaigns: ', error);
+        }
+    };
+
+    const extractTextFromLink = (link) => {
```

Comment:
It's not the same case because we are not getting elements directly from the DOM, we already are getting it from the `campaigns.names` but the list table component sends these data as URLs, so we are using this function just to remove the `<a>` and `</a>` tags from the `campaigns.names` entries. See: https://github.com/impress-org/givewp/pull/7612/files#diff-7b60bf2e4dc0b8c9c1179bb0789dad09951bbf2872af5c210220889b75901722R154

Reply:
I wonder why we are sending links in the first place, that's strange. The ListTable component is messed up. But, you are right, it is not the same.

Reply:
It works that way because the items can be clickable in the confirmation modals from the list table - check the attached screenshot for reference. 

![image](https://github.com/user-attachments/assets/4949ab0d-ff19-4b1c-93a2-42daec4bf0fa)